light_curves/code_src/hats_functions.py

```diff
@@ -0,0 +1,133 @@
+import numpy as np
+import pandas as pd
+from astropy.table import Table
+import lsdb
+from dask.distributed import Client
+from data_structures import MultiIndexDFObject
+from upath import UPath
+
+def hats_get_lightcurves(
+    sample_table: Table,
+    object_catalog_path: str,
+    lightcurve_catalog_path: str,
+    object_columns: list[str],
+    light_curve_columns: list[str],
+    filter_id_to_name: dict[int, str],
+    id_col: str,
+    time_col: str,
+    flux_col: str,
+    err_col: str,
+    *,
+    radius: float = 1.0,
+    object_margin_cache=None,
+    lightcurve_margin_cache=None,
```

Comment:
I think this margin arg is unnecessary since this catalog is not being crossmatched, only joined on object ID.